contentcuration/contentcuration/static/js/edit_channel/tree_edit/views.js

```diff
@@ -541,7 +543,7 @@ var ContentItem = BaseViews.BaseWorkspaceListNodeItemView.extend({
 		if(this.model.has_related_content()){
 			dialog.alert(this.get_translation("warning"), this.get_translation("related_content_alert"), this.copy_item);
 		} else {
-			this.copy_item();
+			this.copy_item(null, "button in tree view");
```

Comment:
Wondering if it might be useful to use the element id or have a set library of tracking strings so that it's easier for another developer to add to the analytics in a consistent way? Maybe something like:
```
ANALYTICS_STRINGS = {
     "#main .copy_button": "Tree View Copy Button",
    "#queue .copy_button": "Clipboard Copy Button",
}
```

Just a thought, no need to address now

Reply:
Wow, I think this is a brilliant idea and definitely worthy of consideration.

Some thoughts:

- how to add custom event parameters?
- how about non-click events?

I want to see how people first use the analytics util function in this PR before doing something like this, but I really do think it's brilliantly simple and could help with adoption and consistency.